41.md

```diff
@@ -0,0 +1,71 @@
+NIP-41
+======
+
+Authentication of clients to relays
```

Comment:
The authentication is not of a client but of a pubkey.
The nip as is contains a commitment to the relay and the pubkey but not to the client.

In theory, a proxy that gets to read the communication could take over the authenticated connection and run queries using the authentication.

Alice (Client) <--WS--> Eve (proxy) <--WS--> Bob (Relay)

Should we make sure of end-to-end encryption before trusting anything that reaches the relay via this websocket?

```suggestion
Pubkey authentication to relays
```